de-lib/src/main/java/org/iplantc/de/communities/client/presenter/ManageCommunitiesPresenterImpl.java

```diff
@@ -0,0 +1,464 @@
+    private CollaboratorsServiceFacade collaboratorsServiceFacade;
+    private GroupAutoBeanFactory factory;
+    private AppMetadataServiceFacade metadataServiceFacade;
+    private AppAutoBeanFactory appAutoBeanFactory;
+    private UserInfo userInfo;
+    private CollaboratorsUtil collaboratorsUtil;
+    private ManageCommunitiesView view;
+    private AppSelectionDialog appSelectView;
+    private AppsView.Presenter appsPresenter;
+    private ReactSuccessCallback selectAppsCallback;
+    @Inject AsyncProviderWrapper<RetagAppsConfirmationDialog> retagAppsConfirmationDlgProvider;
+
+    @Inject
+    public ManageCommunitiesPresenterImpl(GroupServiceFacade serviceFacade,
+                                          AppUserServiceFacade appUserServiceFacade,
+                                          CollaboratorsServiceFacade collaboratorsServiceFacade,
+                                          GroupAutoBeanFactory factory,
+                                          AppMetadataServiceFacade metadataServiceFacade,
+                                          AppAutoBeanFactory appAutoBeanFactory,
+                                          UserInfo userInfo,
+                                          CollaboratorsUtil collaboratorsUtil,
+                                          ManageCommunitiesView view,
+                                          AppsView.Presenter appsPresenter) {
+        this.serviceFacade = serviceFacade;
+        this.appUserServiceFacade = appUserServiceFacade;
+        this.collaboratorsServiceFacade = collaboratorsServiceFacade;
+        this.factory = factory;
+        this.metadataServiceFacade = metadataServiceFacade;
+        this.appAutoBeanFactory = appAutoBeanFactory;
+        this.userInfo = userInfo;
+        this.collaboratorsUtil = collaboratorsUtil;
+        this.view = view;
+        this.appsPresenter = appsPresenter;
+    }
+
+    @Override
+    public void go(HasOneWidget container, String baseId) {
+        Subject selfSubject = getSelfSubject();
+        Splittable subjectSpl = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(selfSubject));
+
+        ReactCommunities.CommunitiesProps props = new ReactCommunities.CommunitiesProps();
+        props.parentId = baseId;
+        props.presenter = this;
+        props.collaboratorsUtil = collaboratorsUtil;
+        props.currentUser = subjectSpl;
+
+        view.show(props);
+
+        container.setWidget(view);
+    }
+
+    Subject getSelfSubject() {
+        Subject subject = factory.getSubject().as();
+        subject.setName(getFullName());
+        subject.setId(userInfo.getUsername());
+        subject.setDisplayName(subject.getName());
+
+        return subject;
+    }
+
+    String getFullName() {
+        String firstName = userInfo.getFirstName();
+        String lastName = userInfo.getLastName();
+        String name = Strings.isNullOrEmpty(firstName) ? "" : firstName;
+        name += Strings.isNullOrEmpty(lastName) ? "" : " " + lastName;
```

Comment:
If I remember right, I tried to switch to using `name` or `fullName` back when I was first working on collaborator lists and QA complained about some broken tests because of it.  I don't recall if the formatting in that field was just different or if that field was actually blank (maybe for accounts created prior to the user portal or something like that?).  I'll try to look into it though.  If it only broke a few QA tests, then I agree switching to full name field would be much better.

Reply:
My branch was so far behind with all the package and package-lock updates, I ended up needing to do a force-push, but I think I addressed most of your comments in my latest commits.  I still need to look into this particular bit about the name/fullname.  

I also completely spaced on formik because I figured this form was too small for it.  I'll try to refactor to formik outside of this PR though.

Reply:
Yeah, I figure you have the logic working for this form already, and it is pretty small, so no need to refactor it for this PR. 👍